### DISCUSSION TITLE
What's the alternative to `consumeAllItemsAndroid` now that it's removed?

### DISCUSSION THREAD

**Discussion by mars-lan** [hyochan/react-native-iap#1458] [Q&A]
I've been using `consumeAllItemsAndroid` to reset the purchase history of non-consumable items. It's not very clear what's the alternative after its removal in [7.2.0](https://github.com/dooboolab/react-native-iap/releases/tag/7.2.0). `flushFailedPurchasesCachedAsPendingAndroid` doesn't sound like the right method?

**Comment by hyochan** [comment]
Very good question. Actually, I thought this is not a good solution to clear all the history in the batch, and thus, `consuming` items should be handled with `consumables` and not `non-consumable`.

**Reply by mars-lan** [reply]
So what's the solution if we'd like to reset `non-consumeable`? This was a very useful function for testing.

**Reply by hyochan** [reply]
Oh for testing 🤔 @andresesfm Do you think we should add `consumeAllItemsAndroid` back?

**Reply by andresesfm** [reply]
Hello @mars-lan Correct me if I'm not seeing what you describe, but `flushFailedPurchasesCachedAsPending`:
https://github.com/dooboolab/react-native-iap/commit/b527d2bcd0678f42ffe80fd3e2846b58622ef226#diff-032a8a05dd68f689941c5cae1197a606b6fefe2b27085db43aed60c27e92ababL218
and `refreshItems`: https://github.com/dooboolab/react-native-iap/commit/b527d2bcd0678f42ffe80fd3e2846b58622ef226#diff-032a8a05dd68f689941c5cae1197a606b6fefe2b27085db43aed60c27e92ababL198 were both being used for consumables.

**Reply by andresesfm** [reply]
Note, I removed them because one was marked as deprecated and looked pretty much like the other one. If there's a difference please point it out and we can bring it back

**Reply by mars-lan** [reply]
@andresesfm As pointed out by your links, the now-removed `refreshItems` native method (called by`consumeAllItemsAndroid`) is **not** the same as  `flushFailedPurchasesCachedAsPending`. The latter will only consume purchase with the state `PENDING` where as the former will cosume them all, regardless of the state.

Just trying to reiterate the goal here. During testing, I'd like to reset all preivous purchases of non-consumable items in order to go through the purcase flow again. This is only possible with the old `consumeAllItemsAndroid` method, not `flushFailedPurchasesCachedAsPendingAndroid`.

**Reply by mars-lan** [reply]
Ping @andresesfm?

**Reply by mars-lan** [reply]
Ping @andresesfm again?

**Reply by drbarto** [reply]
@andresesfm @mars-lan Any news about this issue? I ran into it as well when upgrading some old code...

**Reply by andresesfm** [reply]
Sorry. I didn't see the replies. It it's needed for testing, why not just iterate over the purchases and consumer them?

**Reply by drbarto** [reply]
@andresesfm thanks for the response :) I wasn't aware that this is possible. After looking through the samples, I assume that this is what you mean (working with version 12.10.5):

```js
const purchases = await RNIap.getAvailablePurchases()
await Promise.all(purchases.map((purchase) => RNIap.finishTransaction({ purchase })))
```

Can you confirm?

**Reply by andresesfm** [reply]
Yes
